helix-view/src/document.rs

```diff
@@ -1034,6 +1091,72 @@ impl Default for Document {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Debug, Default)]
+pub struct Config {
+    pub encoding: Option<&'static encoding::Encoding>,
+    pub line_ending: Option<LineEnding>,
+    pub indent_style: Option<IndentStyle>,
+    pub tab_width: Option<usize>,
+}
```

Comment:
Should this be under the file `helix-view/src/editor.rs` since it is editorconfig?

Reply:
It's been a while since I wrote that, but I think there was a weird reason for writing that. Something about supporting other forms of document configuration. Which, relatedly, is why that's in `document` and not `editor`: because although it's *editor*config, those are configuration options for one document (as specified by editorconfig).